openhands-tools/openhands/tools/delegate/definition.py

```diff
@@ -0,0 +1,143 @@
+"""Delegate tool definitions for OpenHands agents."""
+
+from collections.abc import Sequence
+from typing import TYPE_CHECKING, Literal
+
+from pydantic import Field
+from rich.text import Text
+
+from openhands.sdk.llm.message import ImageContent, TextContent
+from openhands.sdk.tool.tool import (
+    Action,
+    Observation,
+    ToolAnnotations,
+    ToolDefinition,
+)
+
+
+if TYPE_CHECKING:
+    pass
+
+
+class DelegateAction(Action):
+    """Action for delegating tasks to sub-agents."""
+
+    operation: Literal["spawn", "send", "close"] = Field(
+        description="The delegation operation to perform"
+    )
+    task: str | None = Field(
+        default=None, description="Task description for spawn operation"
+    )
+    sub_conversation_id: str | None = Field(
+        default=None,
+        description="ID of the sub-conversation for send/close operations",
+    )
+    message: str | None = Field(
+        default=None, description="Message to send to sub-agent (for send operation)"
+    )
+
+    @property
+    def visualize(self) -> Text:
+        """Return Rich Text representation of this action."""
+        content = Text()
+        content.append(f"Delegate {self.operation}:\n", style="bold blue")
+
+        if self.operation == "spawn" and self.task:
+            content.append(f"Task: {self.task}")
+        elif self.operation == "send" and self.message and self.sub_conversation_id:
+            content.append(f"To {self.sub_conversation_id}: {self.message}")
+        elif self.operation == "close" and self.sub_conversation_id:
+            content.append(f"Sub-agent: {self.sub_conversation_id}")
+
+        return content
+
+
+class DelegateObservation(Observation):
+    """Observation from delegation operations."""
+
+    operation: Literal["spawn", "send", "close"] = Field(
+        description="The delegation operation that was performed"
+    )
+    success: bool = Field(description="Whether the operation was successful")
+    sub_conversation_id: str | None = Field(
+        default=None, description="ID of the sub-conversation (for spawn/send/close)"
+    )
+    message: str = Field(description="Result message from the operation")
+    content: Sequence[TextContent | ImageContent] = Field(
```

Comment:
Why would child agent return ImageContent? seems like not supported now. Can we just keep it TextContent